eoxserver/services/ows/wps/processes/get_height_profile.py

```diff
@@ -0,0 +1,211 @@
+            raise InvalidInputValueError(
+                "coverage", "Invalid coverage name '%s'!" % coverage
+            )
+        try:
+
+            data_items = model.arraydata_items.all()
+
+        except model.arraydata_items.all().length > 1:
+            raise InvalidInputValueError(
+                "coverage", "coverage '%s' has more than one imagery, the profile process handles single images!" % coverage
+            )
+        data_item = data_items[0]
+        ds = gdal_open(data_item, False)
+
+        proj_wkt = 'PROJCS["unknown",GEOGCS["unknown",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]]],PROJECTION["Two_Point_Equidistant"],PARAMETER["Latitude_Of_1st_Point",{}],PARAMETER["Longitude_Of_1st_Point",{}],PARAMETER["Latitude_Of_2nd_Point",{}],PARAMETER["Longitude_Of_2nd_Point",{}],PARAMETER["false_easting",0],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH]]'.format(line[1], line[0], line[3], line[2])
+
+        bbox = ((-1 * line_distance / 2), (100*0.5), line_distance / 2, -(100*0.5))
+        interval = int(interval)
+
+        # # Calculate the number of samples in the profile.
+        num_samples = int(line_distance / interval)
+
+        tmp_ds = '/vsimem/%s' % uuid4().hex
+
+        profile_ds = gdal.Warp(tmp_ds, ds, dstSRS=proj_wkt, outputBounds=bbox,
+                        height=1, width=num_samples, resampleAlg=interpolation_method,
+                        format='Gtiff')
+
+        array_out = profile_ds.GetRasterBand(1).ReadAsArray()
+
+        y = []
+        for (d, value) in enumerate(array_out[0, :]):
+            y.append(value)
+        x = np.arange(0, (num_samples * interval)/1000, interval/1000)
+
+        if (profile['mime_type'] == "text/csv"):
+            _output = CDAsciiTextBuffer()
+            if getattr(_output, 'mime_type', None) is None:
+                setattr(_output, 'mime_type', 'text/csv')
+
+            writer = csv.writer(_output, quoting=csv.QUOTE_ALL)
+            header = ["distance", "elevation"]
+            writer.writerow(header)
+
+            for (d, value) in enumerate(array_out[0, :]):
+                writer.writerow([
+                    d * interval,
+                    value
+                ])
+
+        else:
+            if profile['mime_type'] == "image/png":
+                extension = ".png"
+
+            elif profile['mime_type'] == "image/jpeg":
+                extension = ".jpg"
+
+            elif profile['mime_type'] == "image/tiff":
+                extension = ".tif"
+
+            tmppath = '{}.{}'.format(uuid4().hex, extension)
+            output_filename = 'height_profile.%s' % extension
+            fig = plt.figure()
+            plt.plot(x, y)
+            fig.savefig(tmppath)
```

Comment:
I think it should be possible to not store the output figure on disk but to store it to a `BytesIO` instead. This would help us reduce the overhead and minimize necessary cleanup.